src/txn/txn.c

```diff
@@ -1231,8 +1231,20 @@ __txn_resolve_prepared_op(WT_SESSION_IMPL *session, WT_TXN_OP *op, bool commit,
       (upd->type != WT_UPDATE_TOMBSTONE ||
         (upd->next != NULL && upd->durable_ts == upd->next->durable_ts &&
           upd->txnid == upd->next->txnid && upd->start_ts == upd->next->start_ts));
-    first_committed_upd_in_hs =
-      first_committed_upd != NULL && F_ISSET(first_committed_upd, WT_UPDATE_HS);
+    /*
+     * If the first committed update after the prepared update has already been marked to be
```

Comment:
Shall we make it clear that this comment provides the reasoning for the `WT_UPDATE_TO_DELETE_FROM_HS ` flag check? 

Also, shouldn't the comment say:
` If the first committed update **before** the prepared update... `

Reply:
It should be first committed update older than the prepared upate

Reply:
Yup, that is better. Thanks for the update.